core/vm/evm.go

```diff
@@ -137,9 +137,6 @@ func NewEVM(blockCtx BlockContext, txCtx TxContext, statedb StateDB, chainConfig
 		chainConfig: chainConfig,
 		chainRules:  chainConfig.Rules(blockCtx.BlockNumber, blockCtx.Random != nil, blockCtx.Time),
 	}
-	if txCtx.Accesses == nil && chainConfig.IsPrague(blockCtx.BlockNumber, blockCtx.Time) {
-		evm.Accesses = evm.StateDB.(*state.StateDB).NewAccessWitness()
-	}
```

Comment:
this stuff is recreated in `applyTransaction` and isn't needed before, so this is a redundant creation.

Reply:
I've checked all references to `NewEVM` and there're many places that call this method that aren't `applyTransaction`.

I feel it would be much safer to maybe move [that initialization](https://github.com/gballet/go-ethereum/blob/kaustinen-with-shapella/core/state_processor.go#L123) inside `NewEVMTxContext(...)`? I'm not sure why it makes sense to initialize one of its attributes outside the ctor creating these kind of situations.

Also noted that [that referenced line](https://github.com/gballet/go-ethereum/blob/kaustinen-with-shapella/core/state_processor.go#L123) doesn't do the `chainConfig.IsPrague(blockCtx.BlockNumber, blockCtx.Time)` style check that we did here. I guess that's just creating a little extra garbage for block ranges that the witness doesn't apply, but just mentioning.

If we're always fine with doing it ignoring the right fork (as done today in `applyTransaction`), then moving it to `NewEVMTxContext` is possible since that method doesn't have block num.